### ISSUE TITLE
dist/rollup.js contains ES6 code - by design?

### ISSUE THREAD

**Issue by kzc** [rollup/rollup#1036]
I thought the generated `dist/rollup.js` file was supposed to be ES5. Or is it intentionally ES6 code by design?

Was noticed while debugging #1032. `let` statements are converted to `var`, but `for-of` loops remain.

```
$ grep 'for\s*(\s*.* of ' dist/rollup.js

	for ( var item of map.entries() ) {
	for ( var key of this.keys ) {
				for ( var child of value ) {
	for ( var key of this.keys ) {
				for ( var child of value ) {
			for ( var param of this.params ) {
				for ( var name of extractNames( param ) ) {
	for ( var value of values ) {
		for ( var node of this.body ) {
		for ( var node of this.body ) {
			for ( var node of this.body ) {
	for ( var node of body ) {
	for ( var node of values ) {
		for ( var proxy of node.declarations[0].proxies.values() ) {
			// kind of monster would write `for ( foo.bar of thing ) {...}`
			for ( var name of extractNames( node ) ) {
	for ( var node of ast.body ) {
	for ( var key of keys$1[ raw.type ] ) {
	for ( var node of this.ast.body ) {
	for ( var node of this.ast.body ) {
	for ( var node of this.ast.body ) {
	for ( var node of this.ast.body ) {
```

**Comment by kzc** [comment]
Somewhat related: the version of `buble` used in `rollup` is a bit old...
```
$ node_modules/.bin/buble --version
Bublé version 0.12.5
```
The latest published version is `0.14.0`, with some more PRs queued up in the gitlab project.

**Comment by Rich-Harris** [comment]
It's being [transpiled down to code that will run in Node 0.12](https://github.com/rollup/rollup/blob/master/rollup.config.js#L24-L29) – for-of statements are okay, but block scoping isn't. So I'll close this as things are working as expected (I know, it's kinda surprising that for-of was supported before `let`).

Re dependencies – yeah, there are few that are out of date. Probably time for a spring clean
